### PR TITLE
feat(news-ticker): Add component - FRONT-2515

### PR DESCRIPTION
**Pull request by papegaill**
Previews:
- EC: https://ecl-preview-2199--europa-component-library.netlify.app/playground/ec/?path=/story/components-news-ticker--default
- EU: https://ecl-preview-2199--europa-component-library.netlify.app/playground/eu/?path=/story/components-news-ticker--default

Specs:
- EC: https://citnet.tech.ec.europa.eu/CITnet/confluence/pages/viewpage.action?spaceKey=ECL&title=EC+Ticker
- EU: https://citnet.tech.ec.europa.eu/CITnet/confluence/pages/viewpage.action?spaceKey=ECL&title=EU+Ticker

Points still to be seen:
- ~~play and pause icon must still be provided by the designers~~
- ~~need to investigate more on storybook loading stats, sometime the ticker doesn’t initialise correctly (ok in storybook fullscreen)~~
- ~~have to see with the designers to always keep the same height for the component. otherwise the content below the ticker will dance~~
- ~~documentation comp image missing for the following page: https://ecl-preview-2199--europa-component-library.netlify.app/ec/components/~~

### CODE REVIEW

File: src/implementations/twig/components/news-ticker/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "name": "@ecl/twig-component-news-ticker",
+  "author": "European Commission",
+  "license": "EUPL-1.2",
+  "version": "3.0.2",
+  "description": "ECL News ticker",
+  "publishConfig": {
+    "access": "public"
+  },
+  "dependencies": {
+    "@ecl/twig-component-button": "^3.0.2",
```
Comment (planctus) [inline src/implementations/twig/components/news-ticker/package.json]: would you update this component and its dependencies to the current ECL version, 3.0.3?

File: src/implementations/vanilla/components/news-ticker/news-ticker-print.scss [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/*
+ * News ticker
+ * @define news-ticker
+ */
+
+@use "sass:map";
```
Comment (planctus) [inline src/implementations/vanilla/components/news-ticker/news-ticker-print.scss]: i don't think we need to `@use` anything here.. :)

File: src/implementations/vanilla/components/news-ticker/package.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "@ecl/vanilla-component-news-ticker",
+  "author": "European Commission",
+  "license": "EUPL-1.2",
+  "version": "3.0.2",
```
Comment (planctus) [inline src/implementations/vanilla/components/news-ticker/package.json]: same as for the twig component, not really necessary, but you can align the version and the dependencies to 3.0.3

File: src/presets/dev/package.json [hunk 4]
```diff
@@ -38,6 +38,7 @@
     "@ecl/vanilla-component-link": "^3.0.3",
     "@ecl/vanilla-component-menu": "^3.0.3",
     "@ecl/vanilla-component-message": "^3.0.3",
+    "@ecl/vanilla-component-news-ticker": "^3.0.2",
```
Comment (planctus) [inline src/presets/dev/package.json]: align it also here..

File: src/resources/icons-eu/src/all/plause-filled.svg [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (planctus) [inline src/resources/icons-eu/src/all/plause-filled.svg]: i think there is a typo in the file name, "plause-filled"..

File: src/resources/icons-ec/src/all/plause.svg [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (planctus) [inline src/resources/icons-ec/src/all/plause.svg]: is that on purpose named "plause" and not "pause"..? :)

File: src/specs/components/news-ticker/package.json [hunk 7]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "@ecl/specs-component-news-ticker",
+  "author": "European Commission",
+  "license": "EUPL-1.2",
+  "version": "3.0.2",
```
Comment (planctus) [inline src/specs/components/news-ticker/package.json]: also here we can align this package with the others

File: src/resources/icons-ec/src/all/plause.svg (hunk 6 above)
Comment (papegaill) [reply src/resources/icons-ec/src/all/plause.svg]: Thks for this !

File: src/website/src/pages/ec/components/news-ticker/index.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+---
+title: Messages
```
Comment (planctus) [inline src/website/src/pages/ec/components/news-ticker/index.md]: i was wondering the reason why i was not finding the news ticker in the menu on the left.. :)

File: src/implementations/vanilla/components/news-ticker/news-ticker.js [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+    const firstSlide = slidesOrigin[0];
+    const lastSlide = slidesOrigin[slidesOrigin.length - 1];
+    const cloneFirst = firstSlide.cloneNode(true);
+    const cloneLast = lastSlide.cloneNode(true);
+
+    // Clone first and last slide
+    this.slidesContainer.appendChild(cloneFirst);
+    this.slidesContainer.insertBefore(cloneLast, firstSlide);
+
+    this.slides = queryAll(this.slideClass, this.element);
```
Comment (planctus) [inline src/implementations/vanilla/components/news-ticker/news-ticker.js:114]: didn't we make this query already in `SlideOrigin`?

Comment (papegaill) [reply src/implementations/vanilla/components/news-ticker/news-ticker.js:114]: I made a small update and added comments, i need to refresh the slides variable after adding new cloned slides.

**Review by emeryro — APPROVED** [review]

**Pull request merged**